src/heliclockter/__init__.py

```diff
@@ -104,6 +115,33 @@ def _validate(cls: Type[DateTimeTzT], v: Any) -> Optional[DateTimeTzT]:
             dt = v if isinstance(v, _datetime.datetime) else parse_datetime(v)
             return cls.from_datetime(dt)
 
+    elif PYDANTIC_V2_AVAILABLE:
+
+        @classmethod
+        def __get_pydantic_core_schema__(cls, _: Any, __: GetCoreSchemaHandler) -> CoreSchema:
+            from_datetime_schema = core_schema.chain_schema(
+                [
+                    core_schema.datetime_schema(),
+                    core_schema.no_info_plain_validator_function(cls.from_datetime),
+                ]
+            )
```

Comment:
We validate with both the internal `datetime_schema` of pydantic, as well as [no_info_plain_validator_function](https://docs.pydantic.dev/latest/api/pydantic_core_schema/#pydantic_core.core_schema.no_info_plain_validator_function), which just takes a function to perform validation, in this case parsing it us as our heliclockter datetime_tz class.